src/app/upgrade/upgrade.component.html

```diff
@@ -28,6 +28,7 @@
       </mat-card-content>
       <mat-card-actions>
         <button mat-raised-button color="primary" routerLink="/">OK</button>
+        <button mat-raised-button color="primary" (click)="upgrade()" *ngIf="enabled || done">Verfiy</button>
```

Comment:
This button just restarts the process.  I think it is confusing, but are you including this to solve the issue of false successes with the upgrade?

Also the spelling is incorrect.  It should be `Verify`

Reply:
It will make user click verify to make sure upgrade was successful.